harness/tests/common/test_schemas.py

```diff
@@ -100,31 +105,46 @@ def run_completeness(self) -> None:
             errors = expconf.completeness_validation_errors(self.case, url)
             if not errors:
                 continue
-            raise ValueError(f"'{self.name}' failed against {url}:\n - " + "\n - ".join(errors))
+            raise ValueError(
+                ("'{}' failed completeness validation against {}:\n - " "\n - ")
+                .format(self.name, url)
+                .join(errors)
+            )
+
+    def run_sanity_errors(self) -> None:
+        if not self.sanity_errors:
+            return
+        self.run_errors(self.sanity_errors)
 
-    def run_errors(self) -> None:
-        if not self.errors:
+    def run_completeness_errors(self) -> None:
+        if not self.completeness_errors:
             return
-        for url, expected in self.errors.items():
+        self.run_errors(self.completeness_errors, test_type="completeness")
+
+    def run_errors(self, error_cases: Any, test_type: Optional[str] = "sanity") -> None:
+        for url, expected in error_cases.items():
             assert isinstance(expected, list), "malformed test case"
-            errors = expconf.sanity_validation_errors(self.case, url)
-            assert errors, f"'{self.name}' matched {url} unexpectedly"
+            if test_type == "sanity":
+                errors = expconf.sanity_validation_errors(self.case, url)
+            else:
+                errors = expconf.completeness_validation_errors(self.case, url)
+            assert errors, f"'{self.name}' {test_type} validated against {url} unexpectedly"
             for exp in expected:
                 for err in errors:
                     if re.search(exp, err):
                         break
                 else:
-                    msg = f"while testing '{self.name}', expected to match the pattern\n"
-                    msg += f"    {exp}\n"
+                    msg = f"while testing '{self.name}' for {test_type}, expected to match the"
+                    msg += f" pattern\n    {exp}\n"
                     msg += "but it was not found in any of\n    "
                     msg += "\n    ".join(errors)
                     raise ValueError(msg)
 
     def run_defaulted(self) -> None:
         if not self.defaulted:
```

Comment:
can you mimic the all-or-nothing assertion for `merge_as/merge_src/merged` here for `default_as/defaulted`?  Just as a little defense against people accidentally writing tests that have a `default_as` without a `defaulted`, which would silently not really run at all.